ci-operator/config/jaegertracing/jaeger-operator/jaegertracing-jaeger-operator-main__jaeger-product-ocp-14-lp-interop.yaml

```diff
@@ -0,0 +1,107 @@
+base_images:
+  cli:
+    name: "4.14"
+    namespace: ocp
+    tag: cli
+  operator-sdk:
+    name: "4.14"
+    namespace: origin
+    tag: operator-sdk
+  os:
+    name: ubi
+    namespace: ocp
+    tag: "8"
+binary_build_commands: LD_FLAGS="-X github.com/jaegertracing/jaeger-operator/pkg/version.version=1.46.0
```

Comment:
Why are we not installing the jaeger-operator from the operatorhub here?

Reply:
The Distributed Tracing operators follow a rolling release model with a single release channel. Our product builds closely align with the upstream project builds, with minimal differences, ensuring smooth integration. We prioritize an upstream-first testing policy, which means any new test cases added to the upstream repository are primarily intended for new features not yet included in our shipped product. Consequently, running these e2e test cases on our current product would result in failures. Our objective is to thoroughly test the upcoming version of the operator, identifying and addressing any potential regressions or functional issues. Once this pull request (PR) is merged, we will incorporate additional jobs to test our upcoming operator version's product builds. We will also implement stage testing to ensure a seamless experience when deploying the latest product build. Additionally, this strategy greatly aids our developers in conducting pre-merge and post-merge testing for their pull requests (PRs). To facilitate this process, we have implemented Prow CI pull jobs specifically designed for this purpose.

Reply:
@ascerra On each operator scenario, we construct the primary operator from upstream and install any necessary dependency operators from the operatorhub required by any specific test cases. This process applies to other operators such as OpenTelemetry and Tempo. 

To streamline the workflow, I propose renaming these jobs as -pre-release-lp-interop and incorporating tests that evaluate released layered products with unreleased OpenShift versions as required by the interop program. I will implement these changes upon returning to work on Monday, as it is currently late in my timezone.

Reply:
I have updated the PR and implemented the following changes:

* Added interop jobs to test released Distributed Tracing operators with unreleased OpenShift versions.
* Included Distributed Tracing  jobs to test unreleased operators with supported OpenShift versions, architecture types, and cluster configurations. Currently, OCP 4.14 is added, and more jobs will be added after this PR is merged.
* Presubmit jobs have already been added for our developers to test their PRs before and after merging.

I have not renamed the previous jobs with '-pre-release-lp-interop' as this is part of the planned interop program and we do not want to hold this PR for merge. There are a number of tasks blocked due to this PR. 

Please review the interop jobs included in this PR and disregard other periodic and presubmit job reviews that are unrelated to interop.